authority/provisioner/ssh_options.go

```diff
@@ -0,0 +1,108 @@
+package provisioner
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.step.sm/crypto/sshutil"
+)
+
+// SSHCertificateOptions is an interface that returns a list of options passed when
+// creating a new certificate.
+type SSHCertificateOptions interface {
+	Options(SignSSHOptions) []sshutil.Option
+}
+
+type sshCertificateOptionsFunc func(SignSSHOptions) []sshutil.Option
+
+func (fn sshCertificateOptionsFunc) Options(so SignSSHOptions) []sshutil.Option {
+	return fn(so)
+}
+
+// SSHOptions are a collection of custom options that can be added to each
+// provisioner.
+type SSHOptions struct {
+	// Template contains an SSH certificate template. It can be a JSON template
+	// escaped in a string or it can be also encoded in base64.
+	Template string `json:"template,omitempty"`
+
+	// TemplateFile points to a file containing a SSH certificate template.
+	TemplateFile string `json:"templateFile,omitempty"`
+
+	// TemplateData is a JSON object with variables that can be used in custom
+	// templates.
+	TemplateData json.RawMessage `json:"templateData,omitempty"`
+}
+
+// HasTemplate returns true if a template is defined in the provisioner options.
+func (o *SSHOptions) HasTemplate() bool {
+	return o != nil && (o.Template != "" || o.TemplateFile != "")
+}
+
+// SSHTemplateOptions generates a CertificateOptions with the template and data
```

Comment:
generates a SSHCertificateOptions, even though that's pretty obvious.

Reply:
Fixed in b7269b6